packages/dom/src/focusable.js

```diff
@@ -56,10 +56,21 @@ function isVisible( element ) {
  * @return {boolean} Whether element should be skipped from focusable elements.
  */
 function skipFocus( element ) {
-	return (
+	if (
 		element.nodeName.toLowerCase() === 'iframe' &&
 		element.getAttribute( 'tabindex' ) === '-1'
-	);
+	) {
+		return true;
+	}
+
+	if (
+		element.classList.contains( 'components-dropdown' ) &&
+		element.getAttribute( 'tabindex' ) === '-1'
+	) {
+		return true;
+	}
```

Comment:
This prevents the wrapper of a `Dropdown` from being treated as focusable. The wrapper has `tabindex="-1"` to work around a browser bug.

https://github.com/WordPress/gutenberg/blob/d4794d6364c496f82c937e7816c24bb89c8d6da1/packages/components/src/dropdown/index.js#L89-L92

Reply:
Could you go into more detail about what the erroneous behaviour that this fixes is?

I don't think the dom package should have a classname like `components-dropdown` in it, as it's just a general purpose library for working with the DOM. Any other way to achieve this?

Reply:
https://github.com/WordPress/gutenberg/pull/31170 goes into why the `tabindex="-1"` was added. 

If we don't prevent it from being treated as focusable then `TreeGrid` will incorrectly move focus to the dropdown container container instead of the dropdown toggle when focus is on the block name and you press <kbd>→</kbd>.

Maybe a better alternative is to ignore `.components-dropdown` in `getRowFocusables`. It does mean this issue will probably come up again though for a different component which uses `focus.focusable.find()`. What do you think?

https://github.com/WordPress/gutenberg/blob/cdc5c152554773c340dbcc386066d1b96a5b0143/packages/components/src/tree-grid/index.js#L26-L36

Reply:
Oh wait, I'm being silly, I need to revert the change to reproduce it 🤦

Reply:
From MDN (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex):
> A negative value (usually tabindex="-1") means that the element is not reachable via sequential keyboard navigation, but could be focused with JavaScript or visually by clicking with the mouse. It's mostly useful to create accessible widgets with JavaScript.

It seems like rather than adding more `if` statements, the existing `if` statement could be broadened.

The difficulty here is that `find` and `skipFocus` are not specific to keyboard navigation. Maybe `focusable` needs a way to retrieve focusable elements for keyboard navigation only, and when doing this it always skips `tabIndex=-1` regardless of classname or other semantics.

Reply:
Oh yeah, that's a good idea. How about `focus.focusable.find( { keyboard: true } )`?

Reply:
I agree. Also slightly off-topic, only checking `tabindex === '-1'` doesn't seem to be enough in the original code. Maybe it should be `element.tabIndex < 0` instead.